src/ir/item.rs

```diff
@@ -277,11 +278,20 @@ impl Item {
                     TypeKind::Named(ref name, _) => {
                         return name.to_owned();
                     }
-                    _ => {}
-                }
-
-                ty.name().map(ToOwned::to_owned)
-                         .unwrap_or_else(|| format!("_bindgen_ty{}", self.id()))
+                    // We really codegen and use the inner type, so use an empty
+                    // base name so codegen doesn't get confused.
```

Comment:
I meant something like "We call codegen on the inner type, and since we use the parent chain to generate the canonical name, and we don't want this to appear, and can't be referenced itself from another type (we point to the actual alias for that), there's no need to generate a name for it".

Does that make sense?